pkg/server/plugin/notifier/gcpbundle/gcpbundle.go

```diff
@@ -0,0 +1,258 @@
+	return p
+}
+
+func (p *Plugin) SetLogger(log hclog.Logger) {
+	p.log = log
+}
+
+func (p *Plugin) BrokerHostServices(broker catalog.HostServiceBroker) error {
+	has, err := broker.GetHostService(hostservices.IdentityProviderHostServiceClient(&p.identityProvider))
+	if err != nil {
+		return err
+	}
+	if !has {
+		return status.Errorf(codes.FailedPrecondition, "IdentityProvider host service is required")
+	}
+	return nil
+}
+
+func (p *Plugin) Notify(ctx context.Context, req *notifier.NotifyRequest) (*notifier.NotifyResponse, error) {
+	config, err := p.getConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	switch req.Event.(type) {
+	case *notifier.NotifyRequest_BundleUpdated:
+		// ignore the bundle presented in the request. see updateBundleObject for details on why.
+		if err := p.updateBundleObject(ctx, config); err != nil {
+			return nil, err
+		}
+	}
+	return &notifier.NotifyResponse{}, nil
+}
+
+func (p *Plugin) NotifyAndAdvise(ctx context.Context, req *notifier.NotifyAndAdviseRequest) (*notifier.NotifyAndAdviseResponse, error) {
+	config, err := p.getConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	switch req.Event.(type) {
+	case *notifier.NotifyAndAdviseRequest_BundleLoaded:
+		// ignore the bundle presented in the request. see updateBundleObject for details on why.
+		if err := p.updateBundleObject(ctx, config); err != nil {
+			return nil, err
+		}
+	}
+	return &notifier.NotifyAndAdviseResponse{}, nil
+}
+
+func (p *Plugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (resp *spi.ConfigureResponse, err error) {
+	if p.identityProvider == nil {
+		return nil, status.Error(codes.FailedPrecondition, "IdentityProvider host service is required but not brokered")
+	}
+
+	config := new(pluginConfig)
+	if err := hcl.Decode(&config, req.Configuration); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to decode configuration: %v", err)
+	}
+
+	if config.Bucket == "" {
+		return nil, status.Error(codes.InvalidArgument, "bucket must be set")
+	}
+	if config.ObjectPath == "" {
+		return nil, status.Error(codes.InvalidArgument, "object_path must be set")
+	}
+
+	p.setConfig(config)
+	return &spi.ConfigureResponse{}, nil
+}
+
+func (p *Plugin) GetPluginInfo(ctx context.Context, req *spi.GetPluginInfoRequest) (*spi.GetPluginInfoResponse, error) {
+	return &spi.GetPluginInfoResponse{}, nil
+}
+
+func (p *Plugin) getConfig() (*pluginConfig, error) {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+	if p.config == nil {
+		return nil, status.Error(codes.FailedPrecondition, "not configured")
+	}
+	return p.config, nil
+}
+
+func (p *Plugin) setConfig(config *pluginConfig) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.config = config
+}
+
+func (p *Plugin) updateBundleObject(ctx context.Context, c *pluginConfig) (err error) {
+	client, err := p.hooks.newBucketClient(ctx, c.ServiceAccountFile)
+	if err != nil {
+		return status.Errorf(codes.Unknown, "unable to instantiate bucket client: %v", err)
```

Comment:
What conditions can cause this error? Bad service account file?

Reply:
Practically speaking that is probably the case.